File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(),skill.end());
        long long ans=0;
        int n=skill.size();
        int team=skill[0]+skill[n-1];

        for(int i=0;i<n/2;i++){
            if(skill[i]+skill[n-i-1]!=team) return -1;
            ans+=skill[i]*skill[n-i-1];
        }
        return ans;
    }
};

File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);
        int n=skill.length;
        long ans=0;
        int tem=skill[0]+skill[n-1];
        for(int i=0;i<n/2;i++){
            // System.out.println(skill[i]+" "+skill[n-i-1]);
            if(tem!=(skill[i]+skill[n-i-1])) return -1;
            ans+=(skill[i]*skill[n-i-1]);
        }
        return ans;
    }
}